extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/KafkaTestSupport.java

```diff
@@ -63,84 +59,45 @@
-
-    private EmbeddedZookeeper zkServer;
-    private String zkConnect;
+    private static final String TEST_KAFKA_VERSION = System.getProperty("test.kafka.version", "7.0.1");
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaTestSupport.class);
     private Admin admin;
 
-    private KafkaServer kafkaServer;
     private KafkaProducer<Integer, String> producer;
     private KafkaProducer<String, String> stringStringProducer;
-    private int brokerPort = -1;
     private String brokerConnectionString;
+    private KafkaContainer kafkaContainer;
 
     public void createKafkaCluster() throws IOException {
-        System.setProperty("zookeeper.preAllocSize", Integer.toString(128));
-        zkServer = new EmbeddedZookeeper();
-        zkConnect = ZK_HOST + ':' + zkServer.port();
-
-        Properties brokerProps = new Properties();
-        brokerProps.setProperty("zookeeper.connect", zkConnect);
-        brokerProps.setProperty("broker.id", "0");
-        brokerProps.setProperty("log.dirs", Files.createTempDirectory("kafka-").toAbsolutePath().toString());
-        brokerProps.setProperty("listeners", "PLAINTEXT://" + BROKER_HOST + ":0");
-        brokerProps.setProperty("offsets.topic.replication.factor", "1");
-        brokerProps.setProperty("offsets.topic.num.partitions", "1");
-        // we need this due to avoid OOME while running tests, see https://issues.apache.org/jira/browse/KAFKA-3872
-        brokerProps.setProperty("log.cleaner.dedupe.buffer.size", Long.toString(2 * 1024 * 1024L));
-        brokerProps.setProperty("transaction.state.log.replication.factor", "1");
-        brokerProps.setProperty("transaction.state.log.num.partitions", "1");
-        brokerProps.setProperty("transaction.state.log.min.isr", "1");
-        brokerProps.setProperty("transaction.abort.timed.out.transaction.cleanup.interval.ms", "200");
-        brokerProps.setProperty("group.initial.rebalance.delay.ms", "0");
-        KafkaConfig config = new KafkaConfig(brokerProps);
-        Time mock = new MockTime();
-        kafkaServer = TestUtils.createServer(config, mock);
-        brokerPort = TestUtils.boundPort(kafkaServer, SecurityProtocol.PLAINTEXT);
+        kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:" + TEST_KAFKA_VERSION))
+                .withEmbeddedZookeeper()
+                .withLogConsumer(new Slf4jLogConsumer(LOGGER));
+        kafkaContainer.start();
 
-        brokerConnectionString = BROKER_HOST + ':' + brokerPort;
+        brokerConnectionString = kafkaContainer.getBootstrapServers();
         Properties props = new Properties();
         props.setProperty("bootstrap.servers", brokerConnectionString);
         admin = Admin.create(props);
     }
 
     public void shutdownKafkaCluster() {
-        if (kafkaServer != null) {
-            kafkaServer.shutdown();
+        if (kafkaContainer != null) {
+            kafkaContainer.stop();
             admin.close();
```

Comment:
We should close `admin` only in case it's not `null`. If non-existing version of Kafka is passed via `test.kafka.version` then `kafkaContainer` is not null but admin is null (since `kafkaContainer.start()` throws exception). It's minor issue but why not to fix it.